From a1299e3312f0866abbdeb30bf33ffd56e9e9ff5c Mon Sep 17 00:00:00 2001
From: t <tristan75015@gmail.com>
Date: Mon, 5 Oct 2026 14:22:18 +0200
Subject: [PATCH 02/21] MFront: put linker flags after the object files

Libraries given in the linker flags (for instance -ltbb) were ignored by
linkers using the --as-needed policy.
---
 mfront/src/MakefileGenerator.cxx | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/mfront/src/MakefileGenerator.cxx b/mfront/src/MakefileGenerator.cxx
index d82b34da8..c50febe33 100644
--- a/mfront/src/MakefileGenerator.cxx
+++ b/mfront/src/MakefileGenerator.cxx
@@ -589,9 +589,6 @@ namespace mfront {
       } else {
         m << sb << cc << " ";
       }
-      if (!ldflags.empty()) {
-        m << "$(LDFLAGS) ";
-      }
       if (o.sys == "win32") {
         m << "-shared -Wl,--add-stdcall-alias,--out-implib,lib" << l.name
           << "_dll.a,-no-undefined ";
@@ -605,6 +602,12 @@ namespace mfront {
       }
       m << "$^  -o $@ ";
       m << getLibraryLinkFlags(t, o, l.name);
+      // the linker flags may contain libraries (for instance `-ltbb`) which
+      // must be given after the object files, otherwise they are ignored by
+      // linkers using the `--as-needed` policy
+      if (!ldflags.empty()) {
+        m << " $(LDFLAGS)";
+      }
       m << "\n";
       if (o.sys == "apple") {
         m << "\tinstall_name_tool -add_rpath  $(shell " << tfel_config
-- 
2.43.0

